sense/src/types.rs

```diff
@@ -0,0 +1,45 @@
+use scale_info::TypeInfo;
+use frame_support::pallet_prelude::*;
+use codec::MaxEncodedLen;
+
+#[derive(Encode, Decode, Default, Eq, Copy, PartialEq, Clone, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct Entity<AccountId, BlockNumber, BoundedString> {
+	account: AccountId,
+	index: u128,
+	cid: BoundedString,
+	created: BlockNumber,
+	mutated: BlockNumber,
+}
+
+impl<AccountId, BlockNumber, BoundedString> Entity<AccountId, BlockNumber, BoundedString> {
+	pub fn new(
+		account: AccountId,
+		block_number: BlockNumber,
+		index: u128,
+		cid: BoundedString,
+	) -> Entity<AccountId, BlockNumber, BoundedString>
+	where
+		BlockNumber: Clone,
+	{
+		Entity { account, index, cid, created: block_number.clone(), mutated: block_number }
+	}
+}
+
+#[derive(Encode, Decode, Default, Eq, Copy, PartialEq, Clone, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct EntityProperty<BlockNumber> {
+	pub(crate) value: u64,
```

Comment:
I would also discuss privacy of `Entity` and `EntityProperty` fields. Is there a reason why they have different visibility scope - `private` vs `pub(crate)`?

Reply:
Indeed, it is not necessary to have public fields there. I created a getter instead, thanks.